doofinder-for-woocommerce/includes/api/class-update-on-save-api.php

```diff
@@ -220,4 +252,31 @@ private function send_request( $url, $data, $method = 'POST' ) {
 		}
 		return false;
 	}
+	public static function build_search_engines( bool $process_all_languages, $language ) {
```

Comment:
Something like this could work:

```suggestion
  /**
   * Builds a list of search engines based on language settings and configuration.
   *
   * This function constructs an array of search engine hashes for each language available 
   * when processing all languages. It retrieves the search engine hash for each language 
   * using the base locale or language code. If `process_all_languages` is false or the API 
   * credentials are missing, it returns early.
   *
   * @param bool  $process_all_languages Determines if all available languages should be processed.
   * @param object $language              An instance of a language handler providing methods:
   *                                      - `get_languages()` to retrieve all available languages.
   *                                      - `get_base_locale()` to get the base locale.
   *
   * @return array|bool An associative array mapping base locales to search engine hashes 
   *                    if processing all languages, or `false` if API credentials are missing.
   */
	public static function build_search_engines( bool $process_all_languages, $language ) {
```